File: app/src/main/java/com/stevecrossin/whatcanicook/entities/RecipeIngredientsTotalDao.java
package com.stevecrossin.whatcanicook.entities;


import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface RecipeIngredientsTotalDao {

    /**
     * Get all recipe ingredients from the recipeIngredientsTotal database.
     */
    @Query("SELECT * FROM recipeingredientstotal;")
    List<RecipeIngredientsTotal> getAllRecipesAndIngredientsTotal();

    /**
     * Adds a list of recipe ingredients from an array into the recipeIngredientsTotal database
     */
    @Insert
    void addRecipeIngredientsTotal(ArrayList<RecipeIngredientsTotal> recipeIngredientsTotals);
}

File: app/src/main/java/com/stevecrossin/whatcanicook/entities/RecipeIngredientsTotalDao.kt
package com.stevecrossin.whatcanicook.entities

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import java.util.*

@Dao
interface RecipeIngredientsTotalDao {

    /**
     * Get all recipe ingredients from the recipeIngredientsTotal database.
     */
    @get:Query("SELECT * FROM recipeingredientstotal;")
    val allRecipesAndIngredientsTotal: List<RecipeIngredientsTotal>

    /**
     * Adds a list of recipe ingredients from an array into the recipeIngredientsTotal database
     */
    @Insert
    fun addRecipeIngredientsTotal(recipeIngredientsTotals: ArrayList<RecipeIngredientsTotal>)
}
